src/main/java/com/commercetools/sync/customers/helpers/CustomerSyncStatistics.java

```diff
@@ -0,0 +1,27 @@
+package com.commercetools.sync.customers.helpers;
+
+import com.commercetools.sync.commons.helpers.BaseSyncStatistics;
+
+import static java.lang.String.format;
+
+public class CustomerSyncStatistics extends BaseSyncStatistics {
+
+    public CustomerSyncStatistics() {
+        super();
+    }
+
+    /**
+     * Builds a summary of the customer sync statistics instance that looks like the following example:
+     *
+     * <p>"Summary: 2 customers have been processed in total (0 created, 0 updated and 0 failed to sync)."
+     *
+     * @return a summary message of the customer sync statistics instance.
+     */
+    @Override
+    public String getReportMessage() {
```

Comment:
I see this method is the same in many other sync statistics classes. What about putting it in the parent class?

Reply:
Every sync instance could have a different message specific to the implementation.

Reply:
I checked and most of them don't have different implementation. It's 4 out of 10 sync instances that have different implementations :) So 6 of them have the same.

Reply:
I don't know where are you seeing value for improvement maybe explain if you see a different case than me. for me, this method just returns a string that is valuable inside its sub classes, maybe you could pass different parts with the constructor but it will look weird  😄

Reply:
So it's like this now. 6 children have this method implemented

```
@Override
    public String getReportMessage() {
        reportMessage = format(
            "Summary: %s types were processed in total (%s created, %s updated and %s failed to sync).",
            getProcessed(), getCreated(), getUpdated(), getFailed());

        return reportMessage;
    }
```

The only difference is the resource name in the message, in this case it is `types`. This could be implemented in the child, but other than that, we can have a common method for all of those 6 children (and more children to come). So then you don't have issues like https://github.com/commercetools/commercetools-sync-java/pull/585#discussion_r497335686.

Also `reportMessage` is a field, which I have no idea why? So I'm for the change to make a mother implementation in `BaseSyncStatistics` and also to remove `reportMessage` field.

Reply:
c61198d